tasks/db_deploy/test/test_db_deploy.py

```diff
@@ -239,33 +253,585 @@ def return_cursor_return_function(conn: connection):
+
+        def mock_get_db_connection_func(inner_db_host: str, inner_db_name: str, inner_db_port: str, inner_db_user: str,
+                                        inner_db_password: str):
+            key = (inner_db_host, inner_db_name, inner_db_port, inner_db_user, inner_db_password)
+            if key in mock_db_connections.keys():
+                return mock_db_connections[key]
+            else:
+                nonlocal mock_get_db_connection_unmocked_call
+                mock_get_db_connection_unmocked_call = key
+                return None
+
+        db_deploy.get_db_connection = mock_get_db_connection_func
+
+        db_created = False
+        mock_create_database_unmocked_call = None
+
+        def mock_create_database_func(inner_connection: connection, inner_drop_database: bool):
+            if inner_connection == creation_connection and inner_drop_database == drop_database:
+                nonlocal db_created
+                db_created = True
+                return True, ''
+            else:
+                nonlocal mock_create_database_unmocked_call
+                mock_create_database_unmocked_call = (inner_connection, inner_drop_database)
+
+        db_deploy.create_database = mock_create_database_func
+
+        db_deploy.inner_task(
+            db_host, db_name, db_port, db_user, db_user_pass, db_admin_pass, drop_database)
+        self.assertTrue(db_created)
+        mock_create_roles_and_users.assert_not_called()
+        creation_connection.close.assert_called_once()
+        mock_create_schema.assert_called_once_with(updating_connection)
+        updating_connection.close.assert_called_once()
+        mock_create_tables.assert_called_once_with(
+            db_host, db_name, db_port, 'postgres', db_admin_pass)
+        self.assertIsNone(mock_get_db_connection_unmocked_call)
+        self.assertIsNone(mock_create_database_unmocked_call)
+        self.assertTrue(db_created)
+
+    @patch('db_deploy.get_db_connection')
+    @patch('db_deploy.create_roles_and_users')
+    @patch('db_deploy.create_schema')
+    @patch('db_deploy.create_tables')
+    @patch('db_deploy.create_database')
+    def test_inner_task_db_did_not_exist(self,
+                                         mock_create_database: MagicMock,
+                                         mock_create_tables: MagicMock,
+                                         mock_create_schema: MagicMock,
+                                         mock_create_roles_and_users: MagicMock,
+                                         mock_get_db_connection: MagicMock):
+        """
+        create_database returns that the db did not exist, requiring a call to create_roles_and_users
+        """
+        db_user_pass = uuid.uuid4().__str__()
+        db_admin_pass = uuid.uuid4().__str__()
+        db_host = uuid.uuid4().__str__()
+        db_name = uuid.uuid4().__str__()
+        db_user = uuid.uuid4().__str__()
+        db_port = uuid.uuid4().__str__()
```

Comment:
May want to use closer to valid data for port. port should be a number. For PostgreSQL default is 5432 but could use a random int converted to string.

Reply:
Makes sense. Fixed.